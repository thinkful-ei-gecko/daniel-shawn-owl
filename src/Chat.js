import React from 'react';
import Message from './Message';



function Chat(props) {
  //Participants & ChatEvents unified so each chat even has all information required to be rendered as message
  const updatedChatEvents = props.ChatEvents.map(chatEvent => {
    const participant = props.Participants.find(p => chatEvent.participantId === p.id);
    chatEvent.participantName = participant.name;
    chatEvent.avatar = participant.avatar;
    chatEvent.time = new Date(chatEvent.timestamp)
    chatEvent.displayMinutes = chatEvent.time.getMinutes()
    chatEvent.displayHours = chatEvent.time.getHours();
    return chatEvent;
  });

    console.log(updatedChatEvents)

  //map touple: chatEvent = element, i = index 
  //we use the index as the key so React doesn't throw a warning
  //not best practice as an ID will always be unique, indexes are only unique until you rerender or reorder, then things can break.
  return( 
    <div className="ChatWindow">
      <ul className='chats'>
      {updatedChatEvents.map((chatEvent, i) => (
      <Message key={i} chatEvent={chatEvent}/>
  ))}
  </ul>
  </div>)
}

export default Chat;