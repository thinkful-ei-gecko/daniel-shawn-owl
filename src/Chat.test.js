import React from 'react'
import ReactDOM from 'react-dom'
import Chat from './Chat'
import renderer from 'react-test-renderer'
import Participants from './Participants'
import ChatEvents from './ChatEvents'


it('Renders without crashing', () =>{
    const div = document.createElement('div');
    ReactDOM.render(<Chat Participants = {Participants} ChatEvents = {ChatEvents}/>, div);
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
        .create(<Chat Participants = {Participants} ChatEvents = {ChatEvents} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});