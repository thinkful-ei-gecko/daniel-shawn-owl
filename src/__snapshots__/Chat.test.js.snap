// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="ChatWindow"
>
  <ul
    className="chats"
  >
    <li>
      <img
        className="Avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <span>
        Axe Kubicka
      </span>
      <span>
        7
        :
        48
      </span>
      <p>
        Hello world
      </p>
    </li>
    <li>
      Caty Flucker
       
      thumbs-up
    </li>
    <li>
      Frank Runciman
       
      thumbs-down
    </li>
    <li>
      Koren Templeton
       
      raise-hand
    </li>
    <li>
      Koren Templeton
       
      clap
    </li>
    <li>
      Koren Templeton
       
      join
    </li>
    <li>
      Ashla Attwool
       
      leave
    </li>
    <li>
      Axe Kubicka
       
      join-stage
    </li>
    <li>
      Axe Kubicka
       
      leave-stage
    </li>
  </ul>
</div>
`;
